src/crimson/os/seastore/async_cleaner.h

```diff
@@ -1130,23 +1129,57 @@ class AsyncCleaner : public SegmentProvider, public JournalTrimmer {
       );
     }
 
+    seastar::future<> io_await_hard_limits() {
+      ceph_assert(!blocking_io);
+      return seastar::do_until(
+        [this] {
+          cleaner.log_gc_state("GCProcess::io_await_hard_limits");
+          return !cleaner.should_block_on_gc();
+        },
+        [this] {
+          blocking_io = seastar::promise<>();
+          return blocking_io->get_future();
+        }
+      );
+    }
+
+    void maybe_wake_background() final {
+      if (is_stopping()) {
+        return;
+      }
+      if (cleaner.gc_should_run()) {
+        do_wake_background();
+      }
+    }
+
+    void maybe_wake_blocked_io() final {
+      if (!cleaner.is_ready()) {
+        return;
+      }
+      if (!cleaner.should_block_on_gc() && blocking_io) {
+        blocking_io->set_value();
+        blocking_io = std::nullopt;
+      }
+    }
+
   private:
     bool is_stopping() const {
       return !process_join;
     }
 
     gc_cycle_ret run();
 
-    void wake() {
-      if (blocking) {
-	blocking->set_value();
-	blocking = std::nullopt;
+    void do_wake_background() {
+      if (blocking_background) {
```

Comment:
I don't see any function or place where this blocking_background is set. Am I correct?